privacy/ccpa/policy.go

```diff
@@ -40,13 +40,17 @@ func (p Policy) Write(req *openrtb.BidRequest) error {
 		req.Regs = &openrtb.Regs{}
 	}
 
+	var err error
+
 	if req.Regs.Ext == nil {
-		req.Regs.Ext = json.RawMessage(`{"us_privacy":"` + p.Value + `"}`)
-		return nil
+		req.Regs.Ext, err = json.Marshal(openrtb_ext.ExtRegs{USPrivacy: p.Value})
+		return err
 	}
 
-	var err error
-	req.Regs.Ext, err = jsonparser.Set(req.Regs.Ext, []byte(`"`+p.Value+`"`), "us_privacy")
+	jsonString, err := json.Marshal(p.Value)
```

Comment:
I'm wondering if it's just better / safer to do a full json.Unmarshal / set the privacy string / json.Marshal. If you feel that's a bit much, I can address it in the future. It would require changing the unit tests from the arbitrary existing json to valid fields of the Reg.Ext.

Reply:
I considered this but came to the conclusion that that would result in any fields that are not defined in `Regs.Ext` being stripped from the JSON, which is probably not what is expected. But I am happy to make this change if you are really sure.

Reply:
Alternatively I could make the change you suggest but unmarshal to a map, so that all fields are preserved. Let me know what you think.

Reply:
Gotcha. I vote for the map approach. I didn't think of that. Good idea.